src/codegen.cpp

```diff
@@ -151,12 +151,12 @@ extern JITEventListener *CreateJuliaJITEventListener();
 bool imaging_mode = false;
 
 // shared llvm state
-JL_DLLEXPORT LLVMContext &jl_LLVMContext = *(new LLVMContext());
+static LLVMContext &jl_LLVMContext = *(new LLVMContext());
```

Comment:
As far as I can tell we still have the assumption of a global context in many places. As an example 

```
static Type *get_pjlvalue(LLVMContext &C) { return T_pjlvalue; }
```

`T_pjlvalue` is uniqued in the global context. Right now it reads a bit odd that lambdas take a LLVM Context, but then use types that have been uniqued.

I suspect until the code is actually made to take arbitrary contexts, LLVM.jl will still need to have access to the `jl_LLVMContext`, since otherwise we can not generate LLVM Modules for llvmcall.

Reply:
Yes, as you can probably tell, these have been structured with eliminating the global context over the next couple of PRs in mind.

Reply:
With the current `llvmcall`, we still need an global context though, ref https://github.com/JuliaLang/julia/pull/25984#issuecomment-593381848. The accessor is not very usable since we need the context in LLVM.jl before GPUCompiler even comes in the picture. So can we just keep it exported for now?

Reply:
At some point, you can't keep blocking progress here, just because you relied in the past on an internal implementation detail. While I have delayed work on this for a couple years, we're getting to the point where there isn't much left to do to flip the switch.

Reply:
> blocking progress

Well, that's a fun characterization of our GPU compiler efforts.

Reply:
> we're getting to the point where there isn't much left to do to flip the switch.

Flipping the switch is fine, but doing so without providing the alternative of importing a serialized bitcode module seems unnecessary, and actively breaks a large portion of the ecosystem.